coldfront/plugins/qumulo/tasks.py

```diff
@@ -163,3 +167,157 @@ def __validate_results(quota_usages, logger) -> bool:
         logger.warn("Unsuccessful ingestion of quota daily usage. Check the results.")
 
     return success
+
+
+class ResetAcl(object):
+    qumulo_api = None
+    sub_allocations = None
+
+    def __init__(self, allocation: Allocation):
+        self.allocation = allocation
+        self.debug = ENV.bool('DEBUG', default=False)
+        self.fs_path = allocation.get_attribute(name='storage_filesystem_path')
+        self.is_allocation_root = QumuloAPI.is_allocation_root_path(
+            self.fs_path
+        )
+        access_allocations = AclAllocations.get_access_allocations(allocation)
+        self.rw_group = AclAllocations.get_allocation_rwro_group_name(
+            access_allocations,
+            'rw'
+        )
+        self.ro_group = AclAllocations.get_allocation_rwro_group_name(
+            access_allocations,
+            'ro'
+        )
+        self.reset_exclude_paths = []
+        if self.is_allocation_root:
+            self.sub_allocations = []
+            links = None
+            try:
+                links = AllocationLinkage.objects.get(parent=allocation)
+            except AllocationLinkage.DoesNotExist:
+                pass
+            if links is not None:
+                for child in links.children.all():
+                    self.reset_exclude_paths.append(
+                        child \
+                            .get_attribute(name='storage_filesystem_path') \
+                            .rstrip('/')
+                    )
+                    self.sub_allocations.append(child)
+        else:
+            self._setup_qumulo_api()
+            self.parent_aces = AclAllocations.get_sub_allocation_parent_aces(
+                allocation,
+                self.qumulo_api
+            )
+
+    # <debugging functions>
```

Comment:
@ris-bmulligan in future, these may be moved to a mixin. On the other hand, we should also agree on how to use different log levels. I would leave this alone; for now however, it should be a good conversation for setting development logging expectations.